high_card.rb

```diff
@@ -1,5 +1,57 @@
+require "pry"
 # Use these two arrays to generate a deck of cards.
-ranks = ["A", 2, 3, 4, 5, 6, 7, 8, 9, 10, "J", "Q", "K" ]
+classes = [2, 3, 4, 5, 6, 7, 8, 9, 10, "J", "Q", "K", "A" ]
 suits = [ "hearts", "spades", "clubs", "diamonds" ]
+$players = []
+p_1 = $players[0]
+p_2 = $players[1]
+p_3 = $players[2]
 
-players = []
+def build_deck(classes, suits)
+  # build a deck
+    # initialize an empty deck
+    deck = []
+    # loop through ranks
+      # .each with index -> to pull the worth of a card
+        Classes.each_with_index do |classes, index|
+          suits.each do |suit|
+            # loops suits
+              # push each pair into the deck
+              card = {
+                suit: suit,
+                class: classes,
+                worth: index
+              }
+              deck << card
+          end
+        end
+      # shuffle the deck
+    return deck.shuffle
+end
+
+def choose_players
+  #prompt user to enter name
+    # or enter play
+      #take each player entered
+        # add to the our players array
+  while true do
+    puts "Hello, pres 'p' to play"
+    name = gets.chomp.downcase #gets rid of extra line
+    players << name
+    break if name == 'p'
+    $players << name
+    #break
+  end
+  return players
```

Comment:
Yes!  Since you are returning the players, you don't need a global variable.  Get rid of it.